### PR TITLE
Add nanosecond resolution time stamping to socketcan

### LINKED ISSUE #938: Could use nansosecond resolution time stamping for Linux sockets
We could use `SO_TIMESTAMPNS` instead of `SO_TIMESTAMP` ins socket-based interfaces like `socketcan` (or maybe the one in #644, where I stumbled upon it). See [here](https://www.kernel.org/doc/Documentation/networking/timestamping.txt) for docs.

### PR DESCRIPTION
**Pull request by felixdivo**
Implements and closes #938.

### CODE REVIEW

File: can/interfaces/socketcan/socketcan.py [hunk 1]
```diff
@@ -494,6 +504,8 @@ def bind_socket(sock: socket.socket, channel: str = "can0") -> None:
 
     :param sock:
         The socket to be bound
+    :param channel:
+        The channel / interface to ind to
```
Comment (hardbyte) [inline can/interfaces/socketcan/socketcan.py:508]:
```suggestion
        The channel / interface to bind to
```

File: doc/interfaces/socketcan.rst [hunk 2]
```diff
@@ -248,7 +247,8 @@ The :class:`~can.interfaces.socketcan.SocketcanBus` specializes :class:`~can.Bus
 to ensure usage of SocketCAN Linux API. The most important differences are:
 
 - usage of SocketCAN BCM for periodic messages scheduling;
-- filtering of CAN messages on Linux kernel level.
+- filtering of CAN messages on Linux kernel level;
+- usage of nanosecond timings form the kernel.
```
Comment (hardbyte) [inline doc/interfaces/socketcan.rst:251]:
```suggestion
- usage of nanosecond timings from the kernel.
```

**Review by hardbyte — APPROVED** [review]
🕐

**Comment by felixdivo** [comment]
@Mergifyio refresh

**Pull request merged**